agent/agent.go

```diff
@@ -91,35 +91,36 @@ func (a *Agent) Run(payload *queue.Work, cancel <-chan bool) error {
 }
 
 func (a *Agent) prep(w *queue.Work) (*yaml.Config, error) {
-
 	envs := toEnv(w)
 	w.Yaml = expander.ExpandString(w.Yaml, envs)
 
-	// inject the netrc file into the clone plugin if the repository is
-	// private and requires authentication.
+	// append all secrets. they will be checked by the ImageSecrets transform.
 	var secrets []*model.Secret
-	if w.Verified {
-		secrets = append(secrets, w.Secrets...)
-	}
+	secrets = append(secrets, w.Secrets...)
```

Comment:
so before we added the secrets (team and repository) to the list as long as the yaml was verified:

```
var secrets []*model.Secret
if w.Verified { 
   secrets = append(secrets, w.Secrets...)
}
```

this change iterates through the secrets one by one and only adds them to the list if the yaml is verified OR the skip verify is true.

```
   var secrets []*model.Secret

    for _, secret := range w.Secrets {
        if w.Verified || secret.SkipVerify {
                secrets = append(secrets, secret)
        }
    }
```

I'm not sure we need different logic for global variables do we? If the yaml doesn't have an s3 step, and there is a global s3 secret with --image=plugins/s3, it will still be matched in the transform.

At least this is how I'm following the logic. I can take a deeper look tomorrow.